docs/DeveloperGuide.md

```diff
@@ -1,34 +1,148 @@
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## 3. Implementation
+### 3.1 [Proposed] Features
+#### 3.1.1 List By Category
+It extends the current view command.
+The user can use `list` to view all tasks.
+The user can also use `list CATEGORY` to view specific classes whose category is sth. 
+This is the UML design for view by category.
+![UML for View](images/viewCategory.png)
 
+<p>
+Step1: Users add task and add either a default category/new category.
+If user adds default category, then the categoryMap in taskList will not be modified.
+If the user adds a new category, the categoryMap wil add one more key whose value is the category and the value is
+the index for this task.
+</p>
 
-## Product Scope
-### Target user profile
 
-{Describe the target user profile}
+Step2:  User view tasks by category.
+If user only type in `list`, then all tasks will be displayed according to the sequence of task
+index. It simply calls the view function.
+If user type in in `list CATEGORY`, and the category is valid, then it will go for the
+returnCategory method.
+If user type in in `list CATEGORY`, and the category is invalid, then it will go for the
+displayCategory method. 
+Whatever the path, the UI will finally be called either display tasks given the index or display
+category. 
 
-### Value proposition
+#### 3.1.2 [Proposed] View month
+##### 3.1.2.1 Proposed Implementation
+The view month mechanism is facilitated by CalendarCommand which extends Command.
 
-{Describe the value proposition: what problem does it solve?}
+Given below is an example usage scenario and how the mechanism behaves at each step.
 
-## User Stories
+1. The user inputs the command word `calendar`. Upon which, the instance of parser will return a CalendarCommand for execution
+1. CalendarCommand initialises with the following variables, with the help from a calendar class containing the necessary methods related to day/date.
+    * month - if user does not input month, it uses computer's current month
+    * startingDay - which day of the week the first day of the months begins on (0-6, where 0 is Monday)
+    * totalDays - how many days in that month
+    * totalWeeks - number of weeks of the month
+  
+1. The `calendar` command calls on TaskList#findTaskDate() for each day of the month to generate the task listing for a particular day. Only the task event and description is added to calendar view. 
+1. The calendar generated is then stored in a string and UI#showUserMesssage() is called to display the calendar. 
+
+The class diagram below shows the relationships between the different classes required by the `calendar` feature.
+![Sequence diagram for CalendarCommand](images/CalendarCommand_class.jpg)
+
+
+The following diagram summarises what happens when a user executes a new `calendar` command:
+![Sequence diagram for CalendarCommand](images/CalendarCommand_sequence.jpg)
```

Comment:
Should there be a semicolon in front of the word CalendarCommand, such that it is `:CalendarCommand` instead?